### ISSUE TITLE
[SDK] matrix-android-sdk2 uses an unspecced `original_event` field from the `/relations` response

### ISSUE THREAD

**Issue by clokep** [vector-im/element-android#7275]
### Description

The [`/relations` endpoint](https://spec.matrix.org/v1.4/client-server-api/#get_matrixclientv1roomsroomidrelationseventid) does not return an `original_event` field, yet the [matrix-android-sdk2 uses it](https://github.com/matrix-org/matrix-android-sdk2/blob/a4a039b04332746f9592d821512fbc87d11e7ee1/matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/relation/RelationsResponse.kt#L25).

We would like to remove support for this from Synapse in < 6 months. See matrix-org/synapse#12930 for more context.

**Comment by clokep** [comment]
From a quick look of consumers, it seems that this is actually used, but I am not 100% certain:

* https://github.com/matrix-org/matrix-android-sdk2/search?q=originalEvent
* https://github.com/vector-im/element-android/search?q=originalEvent

**Comment by bmarty** [comment]
Yes, this is used. We can try to un-use it, I will check if it's possible to do that fast.
